### PR TITLE
Create setup TUI and improve overall UX

### LINKED ISSUE #16: [Enhancement] TUI Setup Interface
I just created a fork of this project, and I'm going to work on a TUI setup interface. I'm partially inspired by [CubedFileManager](https://github.com/supercrafter100/cubedfilemanager) because it has a nice setup TUI.

If you want, you can go ahead an assign me for this and I'll push a pull request soon.

### PR DESCRIPTION
**Pull request by BlockArchitech**
This pull request adds and/or changes the following:

- Create a TUI for initial set-up.
- Added pycaches to .gitignore
- Fixed requirements.txt (#16 comment three)
- Added a little for verbose for easier error reading.
- Added a loader icon which can be used for almost anything.
- Now in ***full color!***

Fixes/Closes #16

### CODE REVIEW

**Comment by BlockArchitech** [comment]
Also, I'd recommend putting an [open-source license in this project.](https://choosealicense.org)

File: main.py [hunk 1]
```diff
@@ -1,15 +1,61 @@
...
+
+client_id=os.getenv("REDDIT_CLIENT_ID")
+client_secret=os.getenv("REDDIT_CLIENT_SECRET")
+username=os.getenv("REDDIT_USERNAME")
+password=os.getenv("REDDIT_PASSWORD")
+
+console.log("[bold green]Checking environment variables...")
+time.sleep(1)
+
+if client_id == "" or client_secret == "" or username == "" or password == "":
```
Comment (JasonLovesDoggo) [inline main.py:35]: if not all(client_id, client_secret, username, password)

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+# This Input is used to ensure the user is sure they want to continue.
+ensureSetupIsRequired = input("Are you sure you want to continue? > ")
+if ensureSetupIsRequired != "yes":
+	console.print("[red]Exiting...")
+	time.sleep(0.5)
+	exit()
+else:
+	# Again, let them know they are about to erase all other setup data.
+	console.print("[bold red] This will overwrite your current settings. Are you sure you want to continue? [bold green]yes/no")
+	overwriteSettings = input("Are you sure you want to continue? > ")
```
Comment (JasonLovesDoggo) [inline setup.py:36]: I would all .casefold() to all of the input checks as somebody might type 'Yes' and it would fail

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	f.write(f'REDDIT_USERNAME="{user}"\n')
+	time.sleep(0.5)
+	f.write(f'REDDIT_PASSWORD="{passw}"\n')
+
+loader.stop()
+
+console.log("[bold green]Setup Complete! Returning...")
+
+# Post-Setup: send message and try to run main.py again.
+os.system("python3 main.py")
```
Comment (JasonLovesDoggo) [inline setup.py:96]: check the os as python3 isn't a valid command on windows

**Review by JasonLovesDoggo — CHANGES_REQUESTED** [review]
overall great, just would recommend system cross-compatibility and .casefold() on some checks
✔

File: setup.py (hunk 2 above)
Comment (Kamushy) [reply setup.py:36]: agreed

File: setup.py (hunk 3 above)
Comment (owengaspard) [reply setup.py:96]: The command `python3` works on Windows as long as you have it in your PATH.

**Review by owengaspard — CHANGES_REQUESTED** [review]
Amazing work! I suggest that, in addition to checking for missing environment variables, there can be either an env variable or a dotfile that allows the program to know that the setup wizard has been run before. 
For example, you can use the `os.path.exists` function to check if `.setup-run-before` exists. If not, it asks if the user wants to run the program. If it does, check for missing environment variables.
```
from os.path import exists

setup_done = exists(.setup-run-before)
```
so on and so fourth

File: main.py (hunk 1 above)
Comment (BlockArchitech) [reply main.py:35]: ah, I'm so used to python2 where this is the only option!

File: setup.py (hunk 2 above)
Comment (BlockArchitech) [reply setup.py:36]: 👍

File: setup.py (hunk 3 above)
Comment (BlockArchitech) [reply setup.py:96]: IIRC it now is, but I'll look again.

**Comment by BlockArchitech** [comment]
I'll get this added, @owengaspard!

**Comment by Shadowles19** [comment]
> I'll get this added, @owengaspard!

Hey can you let me know when you add those! I would love to use this.

**Comment by BlockArchitech** [comment]
Done! @Shadowles19 if you want to try this out, `git clone https://github.com/BlockArchitech/RedditVideoMakerBot` and follow the README in that repository.

**Comment by BlockArchitech** [comment]
@Kamushy, @owengaspard, @JasonLovesDoggo I've added your changes!

**Comment by BlockArchitech** [comment]
Ah, doesn't everyone love merge conflicts? I'll get this fixed @elebumm

**Review by Kamushy — APPROVED** [review]

**Review by owengaspard — APPROVED** [review]

**Review by Lutacon — APPROVED** [review]

**Comment by BlockArchitech** [comment]
Merge conflicts are resolved. Now it's extra ready for review, @elebumm!

**Review by Lutacon — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,14 @@
...
-__pycache__
\ No newline at end of file
+video_creation/__pycache__/
+.setup-done-before
+||||||| 5b39896
+reddit-bot-351418-5560ebc49cac.json
+=======
+reddit-bot-351418-5560ebc49cac.json
+__pycache__
+>>>>>>> 6fc5d2a7377dfe9f65d0d011fc260602527847c9
```
Comment (Lutacon) [inline .gitignore]: Conflict not resolved here.

**Review by Lutacon — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (BlockArchitech) [reply .gitignore]: git always manages to fail me. I'll get this resolved!

**Review by BlockArchitech — COMMENTED** [review]
It does not appear there are any more conflicts.

**Review by Lutacon — APPROVED** [review]

**Comment by BlockArchitech** [comment]
I'm going to add the new variables to the setup assistant soon.

**Comment by BlockArchitech** [comment]
Merge Conflicts fixed again.

**Comment by elebumm** [comment]
@BlockArchitech Incredible PR. I had some issues with some of the merge conflicts. I also added the `load_dotenv` at the beginning because it wasn't recognizing my PR's. 

I will add this in straight away if we can quickly have those resolved.

**Comment by BlockArchitech** [comment]
Thanks @elebumm! I don't see merge conflicts on my phone, can you clarify what you mean?

Thanks!

**Comment by CallumIO** [comment]
@BlockArchitech I've solved them for you don't worry. They weren't major ones.

**Comment by BlockArchitech** [comment]
Thanks @CallumIO!

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+theme = input("Theme? (light or dark) > ")
+console.log("Attempting to save your credentials...")
+loader = Loader("Saving Credentials...", "Done!").start()
+ # you can also put a while loop here, e.g. while VideoIsBeingMade == True: ...
+time.sleep(0.5)
+console.log("Removing old .env file...")
+os.remove(".env")
+time.sleep(0.5)
+console.log("Creating new .env file...")
+with open('.env', 'a') as f:
```
Comment (CallumIO) [inline setup.py:97]: Before appending, you should clear the file - the .env file is created in main and .env.template copied to it, meaning you'd have duplicate values

**Review by CallumIO — CHANGES_REQUESTED** [review]
Looks great, just that one issue.

File: setup.py (hunk 5 above)
Comment (CallumIO) [reply setup.py:97]: Either this or remove the creation of the file from main

**Comment by BlockArchitech** [comment]
alright @CallumIO how's that?

**Review by CallumIO — APPROVED** [review]
Brilliant, thank you for this, will really help simplify setup for users :D

**Comment by BlockArchitech** [comment]
Yeah, no problem!

**Pull request merged**